source/simulator/core.cc

```diff
@@ -1976,6 +1976,22 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
```

Comment:
Reverted back to adding binary input as a nonlinear solver option that stores the read in binary block vector solution into a temporary block vector without ghost cells before copying into the solution block vector. However, me and @egpuckett  realized that using the binary input solver type in combination with particles will require old_solution and perhaps, old_old_solution as well from the first time step. I am planning on writing some kind of initialization code that reads in the starting time step data, t_n into old_old_solution, t_{n+1} as the old_solution and begin the computation from t_{n+2}.